File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/registry/impl/DubboRegisterFactory.java
package org.javamaster.invocationlab.admin.service.registry.impl;

import org.javamaster.invocationlab.admin.service.registry.Register;
import org.springframework.stereotype.Component;

import java.util.List;

/**
 * @author yudong
 */
@Component
public class DubboRegisterFactory extends AbstractRegisterFactory {

    @Override
    public Register create(String cluster) {
        return new ZkRegister(cluster);
    }

    @Override
    public void refreshService(List<String> interfaceNames, String cluster) {
        Register register = get(cluster);
        register.tryPullDataAgain(interfaceNames);
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/registry/impl/DubboRegisterFactory.kt
package org.javamaster.invocationlab.admin.service.registry.impl

import org.javamaster.invocationlab.admin.service.registry.Register
import org.springframework.stereotype.Component

/**
 * @author yudong
 */
@Component
class DubboRegisterFactory : AbstractRegisterFactory() {
    override fun create(cluster: String): Register {
        return ZkRegister(cluster)
    }

    override fun refreshService(interfaceNames: List<String>, cluster: String) {
        val register = get(cluster)
        register.tryPullDataAgain(interfaceNames)
    }
}
